src/data-model-providers/codegen/CodegenDataModelProvider.cpp

```diff
@@ -635,6 +635,25 @@ CHIP_ERROR CodegenDataModelProvider::SemanticTags(EndpointId endpointId, ReadOnl
 
     return CHIP_NO_ERROR;
 }
+#ifdef CONFIG_USE_ENDPOINT_UNIQUE_ID
+
+CHIP_ERROR CodegenDataModelProvider::EndpointUniqueID(EndpointId endpointId, ReadOnlyBufferBuilder<MutableCharSpan> & builder)
+{
+    char epUniqueId[Clusters::Descriptor::Attributes::EndpointUniqueID::TypeInfo::MaxLength()] = { 0 };
+    MutableCharSpan epUniqueIdSpan(epUniqueId);
+    emberAfGetEndpointUniqueIdForEndPoint(endpointId, epUniqueIdSpan);
+
+    char * buffer = static_cast<char *>(Platform::MemoryCalloc(epUniqueIdSpan.size() + 1, sizeof(char)));
+    VerifyOrReturnError(buffer != nullptr, CHIP_ERROR_NO_MEMORY);
+    memcpy(buffer, epUniqueIdSpan.data(), epUniqueIdSpan.size());
+    MutableCharSpan copiedSpan(buffer, epUniqueIdSpan.size());
+
+    ReturnErrorOnFailure(builder.EnsureAppendCapacity(1));
+    ReturnErrorOnFailure(builder.Append(copiedSpan));
```

Comment:
A few questions:

1. Why are we ignoring the return value from `emberAfGetEndpointUniqueIdForEndPoint`?
2. Why are we doing a copy onto the stack and then another onto the heap?  Could we just have the emberAf thing return a span pointing directly at its data and just have the builder reference that span, so we're not doing copies at all?  This might be a question for @andy31415 
3. If we _are_ doing the heap allocation, where is that "+1" in the allocation size coming from?

Reply:
This should take a `MutableCharSpan &` argument and place the value there. Using an array (readonlybuffers are arrays) and of mutable char spans is odd.